wallet/src/key_chain/account_key_chain.rs

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::key_chain::leaf_key_chain::LeafKeyChain;
+use crate::key_chain::{make_account_path, KeyChainError, KeyChainResult};
+use common::address::pubkeyhash::PublicKeyHash;
+use common::address::Address;
+use common::chain::ChainConfig;
+use crypto::key::extended::{ExtendedPrivateKey, ExtendedPublicKey};
+use crypto::key::hdkd::child_number::ChildNumber;
+use crypto::key::hdkd::derivable::Derivable;
+use crypto::key::PublicKey;
+use std::sync::Arc;
+use storage::Backend;
+use wallet_storage::{StoreTxRo, StoreTxRw, WalletStorageRead, WalletStorageWrite};
+use wallet_types::keys::KeyPurpose;
+use wallet_types::{AccountId, AccountInfo, DeterministicAccountInfo};
+
+#[allow(dead_code)] // TODO remove
+/// This key chain contains a pool of pre-generated keys and addresses for the usage in a wallet
+pub struct AccountKeyChain {
+    /// The specific chain this KeyChain is based on, this will affect the address format
+    chain_config: Arc<ChainConfig>,
+
+    /// The account key from which all the addresses are derived
+    account_pubkey: ExtendedPublicKey,
```

Comment:
ConstValue?

Reply:
Is `ConstValue` a marker type? Because it is still possible to change the field if the struct is mut.

Reply:
Yes. It's for devs to realize that a value is meant to be constant over its lifetime.